drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,412 @@
+/* ST Microelectronics LIS2DUX12 3-axis accelerometer driver
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2dux12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2dux12
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/dt-bindings/sensor/lis2dux12.h>
+
+#include "lis2dux12.h"
+
+LOG_MODULE_REGISTER(LIS2DUX12, CONFIG_SENSOR_LOG_LEVEL);
+
+static int lis2dux12_set_odr(const struct device *dev, uint8_t odr)
+{
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_md_t mode = {.odr = odr};
+
+	return lis2dux12_mode_set(ctx, &mode);
+}
+
+static int lis2dux12_set_range(const struct device *dev, uint8_t range)
+{
+	int err = 0;
+	struct lis2dux12_data *data = dev->data;
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_md_t val = { .odr = cfg->odr, .fs = range };
+
+	err = lis2dux12_mode_set(ctx, &val);
+
+	if (!err) {
+		switch (range) {
+		default:
+		case LIS2DUX12_DT_FS_2G:
+			data->gain = lis2dux12_from_fs2g_to_mg(1);
+			break;
+		case LIS2DUX12_DT_FS_4G:
+			data->gain = lis2dux12_from_fs4g_to_mg(1);
+			break;
+		case LIS2DUX12_DT_FS_8G:
+			data->gain = lis2dux12_from_fs8g_to_mg(1);
+			break;
+		case LIS2DUX12_DT_FS_16G:
+			data->gain = lis2dux12_from_fs16g_to_mg(1);
+			break;
+		}
+	}
+
+	return err;
+}
+
+#define FOREACH_ODR_ENUM(ODR_VAL)                                                                  \
+	ODR_VAL(LIS2DUX12_DT_ODR_OFF, 0.0f)                                                        \
+	ODR_VAL(LIS2DUX12_DT_ODR_1Hz_ULP, 1.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_3Hz_ULP, 3.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_25Hz_ULP, 25.0f)                                                  \
+	ODR_VAL(LIS2DUX12_DT_ODR_6Hz, 6.0f)                                                        \
+	ODR_VAL(LIS2DUX12_DT_ODR_12Hz5, 12.50f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_25Hz, 25.0f)                                                      \
+	ODR_VAL(LIS2DUX12_DT_ODR_50Hz, 50.0f)                                                      \
+	ODR_VAL(LIS2DUX12_DT_ODR_100Hz, 100.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_200Hz, 200.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_400Hz, 400.0f)                                                    \
+	ODR_VAL(LIS2DUX12_DT_ODR_800Hz, 800.0f)
```

Comment:
nit: you can probably pull those `\` a bit behind, no need to make lines unnecessarily wide

Reply:
Alright, I shifted these over quite a bit in the latest commit.